### PR TITLE
Renamed FileVersion to AppendVersion in tag helpers that support cache busting

### PR DESCRIPTION
**Pull request by dpaquette**
FileVersion property renamed to AppendVersion in ImageTagHelper, LinkTagHelper and ScriptTagHelper.

asp-file-version attribute renamed to asp-append-version.

Resolves issue #2540

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs [hunk 1]
```diff
@@ -88,7 +88,7 @@ private enum Mode
             /// <summary>
             /// Just adding a file version for the generated urls.
             /// </summary>
-            FileVersion = 0,
+            AppendVersion = 0,
             /// <summary>
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs]: While here please add blank lines within this `enum` to separate the members. Same in `ScriptTagHelper`.

**Comment by dougbu** [comment]
:shipit: let me know if you can't address my minor comment. If so, I'll merge as-is and we'll do that cleanup some other time.

File: src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs (hunk 1 above)
Comment (dpaquette) [inline src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs]: Will do

**Comment by dpaquette** [comment]
Added blank lines as requested

File: src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs [hunk 2]
```diff
@@ -136,8 +138,8 @@ private enum Mode
         /// <remarks>
         /// If <c>true</c> then a query string "v" with the encoded content of the file is added.
         /// </remarks>
-        [HtmlAttributeName(FileVersionAttributeName)]
-        public bool? FileVersion { get; set; }
+        [HtmlAttributeName(AppendVersionAttributeName)]
+        public bool? AppendVersion { get; set; }
```
Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs]: I need a history lesson on this: why is this a nullable bool? Why not just a bool? Does it ever matter whether it was "not set" or "set to false"? I ask because in WebForms no controls ever needed this behavior (ok fine, and nullable didn't exist in .NET 1.0 either, but it was still possible to implement manually).

Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs]: @DamianEdwards

**Comment by Eilon** [comment]
But anyway, the code here looks great. BTW we're working on some stabilization right now so we will likely wait to merge this until next week if that's ok!

File: src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs (hunk 2 above)
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs]: @Eilon IIRC this might be historical / hysterical. Currently determine mode using `TagHelperContext` (its `AllAttributes` collection) and not the properties themselves. So nullable doesn't look mandatory anymore.

But I suggest we clean this up separately if @DamianEdwards agrees.

**Comment by dpaquette** [comment]
No problem waiting. Let me know if there is anything else I need to do.

**Comment by dougbu** [comment]
@dpaquette sorry to say, there is something else you can do. Not urgent (due to the long weekend) but please rebase your fork on our dev branch and fix the conflicts in your commit. No need to create a new PR; just `git push -f` to your server branch. Let me know when it's updated.

tl;dr; Our recent stabilization work and large number of breaking changes means I can no longer automatically rebase your branch. If I were to address the conflicts, it wouldn't be your change anymore. There are ways around this but you fixing the conflicts is simpler.

**Comment by dpaquette** [comment]
No problem. I will do that this weekend

**Comment by dougbu** [comment]
3fe0490
